draft-regext-rdap-extensions.md

```diff
@@ -56,13 +67,27 @@ contain alphanumeric characters and "_" (underscore) characters. This formulatio
 was explicitly chosen to allow compatibility with variable names in programming
 languages and transliteration with XML.
 
-RDAP extension identifiers have no explicit structure and are opaque in that no
-inner-meaning can be "seen" in them.
-
-When in use in RDAP, extension identifiers are prepended to both URL path segments
-and JSON attribute names. In both cases, the extension identifier acts as a namespace
+When in use in RDAP, extension identifiers are prepended to URL path segments,
+URL query parameters, and JSON object member names (herein further referred to 
+as "JSON names").  In all cases, the extension identifier acts as a namespace
 preventing collisions between extension elements.
 
+Additionally, implementers and operators can use the extension identifiers
+to find an extensions definition (via the IANA registry).
+
+RDAP extension identifiers have no explicit structure and are opaque in that no
+inner-meaning can be "seen" in them. This document restricts the syntax
+of RDAP extension identifiers from containing two consecutive "_" (underscore)
+characters and reserves their use for the future definition of structure (such
+as to define a versioning scheme). That is, RDAP extensions MUST NOT define
+an identifier with two consecutive underscore characters ("__") unless explicitly
+adhering to an RFC describing such usage.
+
+RDAP extensions MUST NOT define an extension identifier in which the identifier
+is a prefix of another RDAP extension identifier. For example, if there were
+a pre-existing identifier of "foo_bar", another RDAP extension could not
+define the identifier "foo".
```

Comment:
If there were an existing identifier "foo", is the intent here that "foobar" could not be used?  Conversely, if there were an existing identifier "foobar", is the identifier "foo" able to be used?

Reply:
No. I'll make that clarification.